koku/api/report/serializers.py

```diff
@@ -196,12 +237,14 @@ def validate(self, data):
 class GroupSerializer(BaseSerializer):
     """A base serializer for group-by operations."""
 
-    pass
+    _tagkey_support = True
 
 
 class OrderSerializer(BaseSerializer):
```

Comment:
A consequence of moving everything involving tags into a single common method is that we're creating fields for order by that are not restricted choice fields. That means any value for a tag field is valid here. I think we either need to make use of the ChoiceField that does the checking for us, or write more custom checking logic for order by fields. 

```
http://127.0.0.1:8000/local/v1/reports/aws/costs/?filter[time_scope_value]=-1&group_by[tag:app]=*&order_by[tag:app]=anything
```

passes serialization but gives a 500, blowing up in the query handler. 

```
Traceback (most recent call last):
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/rest_framework/views.py", line 495, in dispatch
    response = self.handle_exception(exc)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/rest_framework/views.py", line 492, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/aberglund/.local/share/virtualenvs/koku-m3d0lar0/lib/python3.6/site-packages/rest_framework/decorators.py", line 55, in handler
    return func(*args, **kwargs)
  File "/Users/aberglund/redhat/cmaas/koku/koku/api/report/aws/view.py", line 117, in costs
    return _generic_report(request, report='costs', provider='aws')
  File "/Users/aberglund/redhat/cmaas/koku/koku/api/report/view.py", line 389, in _generic_report
    output = handler.execute_query()
  File "/Users/aberglund/redhat/cmaas/koku/koku/api/report/aws/aws_query_handler.py", line 146, in execute_query
    query_order_by.extend([self.order])
  File "/Users/aberglund/redhat/cmaas/koku/koku/api/query_handler.py", line 154, in order
    return f'{order_map[self.order_direction]}{self.order_field}'
KeyError: 'anything'
[2019-05-06 14:07:54,634] ERROR: "GET /local/v1/reports/aws/costs/?filter[time_scope_value]=-1&group_by[tag:app]=*&order_by[tag:app]=anything HTTP/1.1" 500 122516
```

Reply:
To be clear, `_init_tagged_fields()` is just instantiating the relevant serializer class. It doesn't control the field choice.

The decision on what field type to use is still controlled by the respective serializer. So, this should be an easy change and probably one that both OrderSerializer and GroupSerializer should implement.

Reply:
Fixed in 5760bfcd3d